src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  readonly shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+type NumberValue = number;
+
+export class Triangle implements Figure {
+  readonly shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: NumberValue = 0,
+    public b: NumberValue = 0,
+    public c: NumberValue = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid triangle dimensions');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid triangle dimensions');
```

Comment:
[CODE STYLE] - The error message here could be more descriptive by explaining the triangle inequality principle, which states that the sum of the lengths of any two sides must be greater than the length of the remaining side.